### ISSUE TITLE
IDE0009 Add 'this' or 'Me' qualification incorrectly reported

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#28509]
Starting with Visual Studio 15.8.0 Preview 3, Visual Studio has started incorrectly reporting that a variable should be prefixes with `this.` when `dotnet_style_qualification_for_method` is set to true and you are using the variable in curly braces.

For example, the following shows up with errors in my IDE for `value` and `foo` inside of the curly braces due to them not being prefix'ed with `this.` which would cause this code to be invalid.

```csharp
public List<String> Foo(String value)
{
    var foo = "foo";
    return new List<string> { value, foo };
}
```

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/288604/ide0009-add-this-or-me-qualification-incorrectly-r-1.html
VSTS ticketId: 644505_
_These are the original issue comments:_
(no comments)
_These are the original issue solutions:_
(no solutions)

**Comment by oatkins** [comment]
Is this the same issue. With 15.8 preview 4 I get this:

![image](https://user-images.githubusercontent.com/10925800/42839882-1705c4a0-89d3-11e8-8372-fd13f50ccf99.png)

Here, IDE0009 is requesting that `this.` be added *even though it's already there!*

**Comment by Paxxi** [comment]
I'm interested in taking care of this one 😄

**Comment by sharwell** [comment]
@Paxxi Awesome, let me know if you have any questions 😄

**Comment by jinujoseph** [comment]
@JieCarolHu close this after validation

**Comment by JieCarolHu** [comment]
Verified on master.

**Comment by Neme12** [comment]
@ChristopherHaws Unfortunately this fix will only be available in 15.9, not 15.8.
